Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+does not absolve the user from planning out their production storage cluster beforehand.
+
+For the purpose of this document, we will consider two simplified use cases to help us make informed
+decisions about Rook and Ceph:
+ * Co-located: User applications co-exist on nodes running Ceph
+ * Disaggregated: Ceph nodes are totally separated from user applications
+
+
+## General Best Practices
+
+This chapter provides an outline of a series of generalized recommendations for best practices:
+* Ceph monitors are more stable on fast storage (SSD-class or better) according to Ceph best
+  practices. In Rook, this means that the `dataDirHostPath` location in the `cluster.yaml` should be
+  backed by SSD or better on MON nodes.
+* Raise the Rook log level to `DEBUG` for initial deployment and for upgrades, as it will help with
+  debugging problems that are more likely to occur at those times.</br>
+  Ensure that the `ROOK_LOG_LEVEL` in operator.yaml equals `debug`.
+* The Kubernetes CSI driver is the preferred default, but ensure in `operator.yaml` that
+  `ROOK_ENABLE_FLEX_DRIVER` remains set to `false`. This is because the FlexVolume driver is in
+  sustaining mode, is not getting non-priority bug fixes, and will soon be deprecated.
+* Ceph’s placement group (PG) auto-scaler module makes it unnecessary to manually manage PGs. We
+  recommend you always set this to `enabled`, unless you have some need to manage PGs manually. In
+  `cluster.yaml`, enable the `pg_autoscaler` MGR module.
+* Rook has the capability to auto-remove Deployments for OSDs which are kicked out of a Ceph
+  cluster. This is enabled by: `removeOSDSIfOutAndSafeToRemove: true`. This means there is less user
+  OSD maintenance and no need to delete Deployments for OSDs that have been kicked out. Rook will
+  automatically clean up the cluster by removing OSD Pods if the OSDs are no longer holding Ceph
+  data. However, keep in mind that this can reduce the visibility of failures from Kubernetes Pod
+  and Pod Controller views. You can optionally set `removeOSDSIfOutAndSafeToRemove` to `false` if
+  need be, such as if a Kubernetes administrator wants to see disk failures via a Pod overview.
+* Configure Ceph using the central configuration database when possible. This allows for more
+  runtime configuration flexibility. Do this using `ceph config set` commands from Rook's toolbox.
+  Only use Rook’s provided ceph.conf override ConfigMap when it is required.
+
+
+## Limiting Ceph to Specific Nodes
+
+One of the more common setups you may want for your Rook-Ceph cluster is to limit Ceph to a specific
+set of nodes. Even for co-located use cases, you could have valid reasons why you must not (or do
+not want to) use some nodes for storage. This is applicable for both co-located and disaggregated
+use cases. To limit Ceph to specific nodes, we can label Kubernetes nodes and configure Rook to have
+Affinity (as a hard preference) for our label.
```

Comment:
I think it's still important to note what the affinity target is.

Reply:
Hmmm then the sentence isn't quite right still because it sounds like it's repeating itself unnecessarily :thinking:  You haven't quite made it clear what the affinity target is and "for our label" doesn't make that clearer. The "for our label" is it referencing the the Kubernetes nodes?

Reply:
"affinity for our labeled nodes" ???

Reply:
:thinking: I think that's probably the best option at the moment.